std/testing/runner.ts

```diff
@@ -168,6 +191,19 @@ export async function runTestModules({
     return;
   }
 
+  const testFile = renderTestFile(testModules);
+  const root = Deno.env("DENO_DIR") || Deno.cwd();
+  const testFilePath = join(root, ".deno.test.ts");
```

Comment:
`Deno.makeTempDir()`? Avoids unnecessary env usage and user space code shouldn't use `DENO_DIR` like this IMO.

Reply:
Actually this is intentional - I should have commented it. 

This actually tricks Deno to not recompile files if there's no need. Let me explain:
1. When you first run `DENO_DIR/.deno.test.ts` Deno will compile it and cache it
2. The file is removed by test runner
3. On next test run file is created again - if you did not add new files that runner picks up and files did not change compiler will recompile them
4. If any of test files changes Deno will pick it up during loading the module into V8 and recompile only this single module

Reply:
Oh of course, the cache would be cluttered with temp modules otherwise.

Reply:
Added a few paragraphs about what's happening there.